test/numerical/Common.hpp

```diff
@@ -1,9 +1,12 @@
-/*
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
-#include <rapidcheck.h>
-#undef _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
-
-#include "test/modellib/ModelLib.hpp"
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"   // gcc, clang
```

Comment:
I only tested this with clang on Mac but according to stackoverflow
https://stackoverflow.com/a/11853015
`#pragma GCC` should work for both gcc and clang